.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,69 @@
+name: Integration Testing  
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  test:
+    name: cs9 x86_64 e2e 
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt-get -y install podman 
+
+      - name: Checkout sources
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          # Use githash of a tested commit instead of merge commit
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: build hirte tests containers
+        run: |
+          podman run --rm --systemd=always --name hirte-builder \
+          -v  ${PWD}:/hirte:Z -it quay.io/yarboa/hirte-builder:1.0.0 \
+          /bin/bash -c "rm -rf builddir; meson setup builddir; \
+          meson install -C builddir --destdir bin"
+          podman build -f tests/Containerfile -t hirte-image .
+
+      - name: run hirte containers
+        run: |
+          podman run -d --rm -p 2020:2020 --systemd=always --name hirte-mgr \
+          $(podman images *hirte-image --format {{.ID}})
+          podman run -d --rm --systemd=always --name hirte-agent-foo \
+          $(podman images *hirte-image --format {{.ID}})
+          podman run -d --rm --systemd=always --name hirte-agent-bar \
+          $(podman images *hirte-image --format {{.ID}})
+          PODMAN_NET=$(hostname -I | awk '{print $1}')
+          echo "PODMAN_NET=$PODMAN_NET" >> $GITHUB_ENV
+
+      - name: run hirte orchestrator
+        run: |
+          podman exec -t  $(podman ps | grep hirte-mgr | awk '{print $1}') \
```

Comment:
Since you set the name of the container, do you really need to look for it in `podman ps`? (same for all cases)

Reply:
Oops. you are correct

Reply:
In some cases you can take advantage of `podman exec -t -l ...`  to exec into the last started container, as long as you are executing commands locally, not on a Mac or Windows platform.